src/EventDispatcher/SimpleEventDispatcher.php

```diff
@@ -11,26 +11,29 @@
 final class SimpleEventDispatcher implements EventDispatcherInterface
 {
     private iterable $listeners;
+
     /** @var object[] */
     private array $events = [];
 
     /**
-     * @param Closure ...$listeners Functions that will handle each event.
+     * @param callable[][] ...$listeners Callables that will handle events.
```

Comment:
I believe this leads to unnecessary complexity of the dispatcher.
This dispatcher does not have to act like a serious production dispatcher.
It doesn't have to call listeners that are individually defined for each event class.
The callable type is not needed here. This is not a replacement for the usual dispatcher, it is just an event interception tool